hazelcast/src/main/java/com/hazelcast/nio/tcp/nonblocking/NonBlockingIOThread.java

```diff
@@ -109,7 +109,7 @@ public final void addTask(Runnable task) {
      * @param task the task to add.
      * @throws NullPointerException if task is null
      */
-    public final void addTaskAndWakeup(Runnable task) {
+    public void addTaskAndWakeup(Runnable task) {
```

Comment:
I'm not quite sure how to test it without making this method non-final. (as final methods can't be mocked)